### PR TITLE
Analysis/endpoints

### PR DESCRIPTION
**Pull request by ceriddenn**
## **Description**

- adding analysis endpoints

## **Issues**

- N/A

---

- [x] I have read the [Contributing Guide](https://docs.waldo.vision/en/getting-started/), and agree to follow the [Code of Conduct](https://docs.waldo.vision/legal/code-of-conduct/).

### CODE REVIEW

**Review by Bensikrac — APPROVED** [review]
as already stated, LGTM, is tested by me aswell

File: apps/web/server/trpc/router/analysisApi.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      const itemCount = await ctx.prisma.gameplay.count();
+      const pagenumber = input.page;
+      const results = await ctx.prisma.gameplay.findMany({
+        skip: pagenumber * 10,
+        take: 10,
+        include: { gameplayVotes: { where: { isGame: true } } },
+      });
+      const noCounts = await ctx.prisma.gameplayVotes.count({
+        where: { isGame: false },
+      });
```
Comment (Huskydog9988) [inline apps/web/server/trpc/router/analysisApi.ts:31]: you might want to consider using a prisma transaction, or using promise.all to limit locking up the thread

File: apps/web/server/trpc/trpc.ts [hunk 2]
```diff
@@ -52,6 +53,56 @@ const isAuthed = t.middleware(async ({ ctx, next }) => {
...
+      message: 'api key invalid',
+      code: 'UNAUTHORIZED',
+    });
+  }
+  if (result.user.blacklisted) {
+    throw new TRPCError({
+      message: 'api key blacklisted',
+      code: 'FORBIDDEN',
+    });
+  }
```
Comment (Huskydog9988) [inline apps/web/server/trpc/trpc.ts:91]: id mark both of these are unauthorized

File: packages/database/prisma/migrations/20230417223139_m3d17/migration.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- DropForeignKey
+ALTER TABLE "Clip" DROP CONSTRAINT "Clip_gameplayId_fkey";
+
+-- AddForeignKey
+ALTER TABLE "Clip" ADD CONSTRAINT "Clip_gameplayId_fkey" FOREIGN KEY ("gameplayId") REFERENCES "Gameplay"("id") ON DELETE CASCADE ON UPDATE CASCADE;
```
Comment (Huskydog9988) [inline packages/database/prisma/migrations/20230417223139_m3d17/migration.sql:5]: don't we already have this migration??

Comment (ceriddenn) [reply packages/database/prisma/migrations/20230417223139_m3d17/migration.sql:5]: I dont believe so… it was never merged. Has always been w this commit. There were two in this branch. One for unique key constraint and the other for deletion cascade.

Comment (Huskydog9988) [reply packages/database/prisma/migrations/20230417223139_m3d17/migration.sql:5]: if you look at https://github.com/waldo-vision/waldo/blob/master/packages/database/prisma/migrations/20230417221847_m4d17/migration.sql it seems to be the same file

Comment (ceriddenn) [reply packages/database/prisma/migrations/20230417223139_m3d17/migration.sql:5]: Ah ok, thats right, i created one then pushed, then bensik took over and created another. I delete one of them when I get home.

**Review by Huskydog9988 — APPROVED** [review]

**Pull request merged**